BlueDBInMemory/src/main/java/io/bluedb/memory/BlueDbInMemory.java

```diff
@@ -1,21 +1,29 @@
 package io.bluedb.memory;
 
 import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
 import io.bluedb.api.BlueDb;
 import io.bluedb.api.exceptions.BlueDbException;
 import io.bluedb.api.BlueCollection;
 
 public class BlueDbInMemory implements BlueDb {
 
+	@SuppressWarnings("rawtypes")
+	Map<Class, BlueCollection> collections = new HashMap<>();
+
+	@SuppressWarnings("unchecked")
 	@Override
 	public <T extends Serializable> BlueCollection<T> getCollection(Class<T> type) {
-		return new BlueCollectionImpl<>(type);
+		if (!collections.containsKey(type)) {
```

Comment:
We don't really save much from only having one instance of each collections right? We just don't have to construct them every time? Having a client use the same collection on different threads isn't a problem right? I don't think it is. Technically this getCollection method could result in multiple collections getting created if its called at the same time from multiple threads. Its really not a problem to create more than one though.